chainer/functions/caffe/caffe_function.py

```diff
@@ -0,0 +1,336 @@
+import collections
+import sys
+
+import numpy
+
+from chainer import function
+from chainer import function_set
+from chainer import functions
+# caffe_pb2 does not support Py3
+if sys.version_info < (3, 0, 0):
+    from chainer.functions.caffe import caffe_pb2
+
+    _type_to_method = {}
+    _oldname_to_method = {}
+
+    def _layer(typ, oldname):
+        def decorator(meth):
+            global _type_to_method
+            _type_to_method[typ] = meth
+            typevalue = getattr(caffe_pb2.V1LayerParameter, oldname)
+            _oldname_to_method[typevalue] = meth
+            return meth
+        return decorator
+else:
+    def _layer(typ, oldname):  # fallback
+        def decorator(meth):
+            return meth
+        return decorator
+
+
+class CaffeFunction(function.Function):
+    """Function using the model file of Caffe.
+
+    Given a binary protobuf file of a Caffe model, this function loads and
+    emulates it on :class:`~chainer.Variable` objects. It supports the official
+    reference models provided by BVLC.
+
+    .. note::
+
+       This function only supports Python 2.7, since the compiled module for
```

Comment:
I change the code to raise an exception in Python 3. I also wrote this behavior in the docstring.